lib/csu/aarch64c/crt1_c.c

```diff
@@ -131,3 +130,20 @@ _start(void *auxv,
 
 	__libc_start1(argc, argv, env, cleanup, main, data_cap, code_cap);
 }
+
+/*
+ * The real entry point _start just sets the unwind info for CRA to undefined
+ * which tells libunwind to stop unwinding and then calls the real __start.
+ * This is needed because ".cfi_undefined" inline assembly in a non-naked
+ * function could be overridden by the default unwinding information.
+ */
+__attribute__((naked, used)) void
+_start(void *auxv,
```

Comment:
What's the benefit of doing this in C rather than just writing assembly? Though I continue to feel that, longer-term, we should push for some GNU attribute that allows you to eschew the assembly entirely, and can make things less stupid upstream.

Reply:
It means I don't need to add another file to the build system. Also means we don't need to be careful about adding all the required extra directives to the .S file.

Reply:
I'm planning to upstream this to FreeBSD as well